meson_options.txt

```diff
@@ -0,0 +1,98 @@
+# Nix project build options
+#============================================================================
+
+# dirs
+#============================================================================
+
+option(
+    'nixstoredir',
+    type : 'string',
+    value : '/nix/store',
+    description : 'path of the Nix store (defaults to /nix/store)')
+
+
+# compiler args
+#============================================================================
+
+option(
+    'ldflags',
+    type : 'array',
+    value : [
+        '-L/usr/local/lib',
+        '-L/usr/lib',
+        '-L/lib'],
+    description : 'Link flags')
+
+
+option(
+    'cxxflags',
+    type : 'array',
+    value : [
+        '-Wno-pedantic',
+        '-Wno-non-virtual-dtor',
+        '-Wno-unused-parameter',
+        '-Wno-variadic-macros',
+        '-Wdeprecated-declarations',
+        '-Wno-missing-field-initializers',
+    ],
+    description : 'C build flags')
+
+
+# optional dependencies
+#============================================================================
+
+option(
+    'enable_gc',
+    type : 'feature',
+    value : 'auto',
+    description : 'Build nix with Boehm garbage collector')
+
+option(
+    'enable_s3',
+    type : 'feature',
+    value : 'auto',
+    description : 'Build nix with s3')
+
+option(
+    'with_coreutils_bin',
+    type : 'string',
+    description : 'path of cat, mkdir, etc.')
+
+
+# misc
+#============================================================================
+option(
+    'doc_generate',
+    type : 'boolean',
+    value : 'false',
+    description : 'Build the documentation and manpages')
+
+# option(
+#     'mandir',
+#     type : 'string',
+#     value : 'share/man',
+#     description : 'Man pages location path')
+
+option(
+    'docdir',
+    type : 'string',
+    value : 'share/doc',
+    description : 'Documentation location path')
+
+option(
+    'build_shared_libs',
```

Comment:
Seems like you could use `default_library` for this.

Reply:
iirc this was only added for parity with the existing stuff

Reply:
most likely not, though library defaults will be more of a faff come time for NT support. i think you could remove this and rely on the meson builtins